test-records.js

```diff
@@ -77,26 +93,76 @@ validator('companies', cdb_schema, (json) => {
+                    errors.push({
+                        msg: `Record specifies '${prop}' of '${json[prop]}' but that isn't available for 'request-language' of '${json['request-language']}'.`,
+                    });
             } else {
                 if (!templates['en'].includes(json[prop]))
-                    fail(
-                        `Record specifies '${prop}' of '${json[prop]}' but that isn't available in English.`,
-                        'See: https://github.com/datenanfragen/data/issues/1120'
-                    );
+                    errors.push({
+                        msg: `Record specifies '${prop}' of '${json[prop]}' but that isn't available in English.`,
+                        ref: 'https://github.com/datenanfragen/data/issues/1120',
+                    });
             }
         }
     }
 
     // A `quality` of `tested` may only be set if `required-elements` are specified (#811).
     if (json['quality'] === 'tested') {
         if (!json['required-elements'])
-            fail(
-                "Record has `quality` of `tested` but doesn't specify `required-elements`.",
-                'See: https://github.com/datenanfragen/data/issues/811'
+            errors.push({
+                msg: "Record has `quality` of `tested` but doesn't specify `required-elements`.",
+                ref: 'https://github.com/datenanfragen/data/issues/811',
+            });
+    }
+    // whitespace check
+    Object.keys(json).forEach((key) => {
+        if (typeof json[key] === 'string' && json[key] !== json[key].trim())
+            errors.push(`Seems like \`${key}\` isn't trimmed, i.e. it contains leading or trailing whitespace.`);
+    });
+
+    // address formatting
+    const address_lines = json['address'].split('\n');
+    if (address_lines.length < 2) errors.push('`address` is not formatted with newlines (\\n).');
+    for (line of address_lines) {
+        if (line !== line.trim()) {
+            ("`address` isn't trimmed, i.e. contains unnecessary whitespace.");
+        }
+    }
+    if (address_lines.includes(json['name'])) errors.push('Record includes `name` in the `address`.');
+
+    // check if the last line is a country
+    // this test might produce false-positives, as it's a difficult thing to do
+    // we might consider more fuzzy matching if we get many reports of false positives
+    const last_line = address_lines[address_lines.length - 1].trim();
+    let last_line_is_country = Object.entries(countries).some((arg) => {
+        // if there is a more elegant way, please let me know
+        [countrycode, ...v] = arg;
+        v = v[0];
+        // we handle US, SG, ... below, as the lib contains it as "United States", but we want "United States of America"
+        if (!['US', 'SG'].includes(countrycode) && v.name == last_line) return true;
+        if (v.native == last_line) return true;
+    });
+    const countrynames = ['United States of America', 'The Netherlands', 'Republic of Singapore'];
```

Comment:
Let's have a clearer name for that. Maybe `country_name_variations` in line with your string below?